pylucene/app/app.py

```diff
@@ -137,7 +137,8 @@ def _get_pred(x):
 
 def run_app():
     uvicorn.run(
-        app,
+        "app.app:app",
         port=8043,
         host='0.0.0.0',
+        reload=True
```

Comment:
Why enable hot reloading if this is supposed to run in Docker anyway? Also, are there any performance or security implications when enabling this?

Reply:
I am enabling it so that It works directly on docker and when ever I update I don't need to restart the docker. I am not sure about the performance and security issues. lets ask @GWNLekkah.

Reply:
I checked, and at the very least the memory consumption will increase. Hence, we should not do this (at least not in prod).

Tbh, in a dev setting it's easier to run without docker anyway

Reply:
Would be a good idea to introduce an ENV variable here for the hot reload, so we could enable it from the docker container instead. I don't agree that it's easier to run without docker in dev, as you would have to do the networking without Traefik. Also, I think it's always nice that your dev environment is closer to the final production environment.

Reply:
Then there's still one thing I don't understand...

How does this even work in Docker? Wouldn't you still need to rebuild the container? It doesn't contain the updated source code right?

Reply:
It contains the updated code, I used volume option and run uvicorn on hot reload.